recipes/libcheck/all/conanfile.py

```diff
@@ -24,96 +29,79 @@ class LibCheckConan(ConanFile):
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["CHECK_ENABLE_TESTS"] = False
+        tc.variables["ENABLE_MEMORY_LEAKING_TESTS"] = False
+        tc.variables["CHECK_ENABLE_TIMEOUT_TESTS"] = False
+        tc.variables["HAVE_SUBUNIT"] = self.options.with_subunit
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy("COPYING.LESSER", src=self._source_subfolder, dst="licenses")
-        cmake = self._configure_cmake()
+        copy(self, "COPYING.LESSER", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
-
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_info(self):
         target = "checkShared" if self.options.shared else "check"
         self.cpp_info.set_property("cmake_file_name", "check")
-        self.cpp_info.set_property("cmake_target_name", "Check::{}".format(target))
+        self.cpp_info.set_property("cmake_target_name", f"Check::{target}")
         self.cpp_info.set_property("pkg_config_name", "check")
 
         # TODO: back to global scope in conan v2 once cmake_find_package_* generators removed
-        libsuffix = "Dynamic" if self._is_msvc and self.options.shared else ""
-        self.cpp_info.components["liblibcheck"].libs = ["check{}".format(libsuffix)]
+        libsuffix = "Dynamic" if is_msvc(self) and self.options.shared else ""
+        self.cpp_info.components["liblibcheck"].libs = [f"check{libsuffix}"]
         if self.options.with_subunit:
             self.cpp_info.components["liblibcheck"].requires.append("subunit::libsubunit")
         if not self.options.shared:
             if self.settings.os in ["Linux", "FreeBSD"]:
                 self.cpp_info.components["liblibcheck"].system_libs = ["m", "pthread", "rt"]
 
+        # TODO: to remove in conan v2
         bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH environment variable: {}".format(bin_path))
+        self.output.info(f"Appending PATH environment variable: {bin_path}")
         self.env_info.PATH.append(bin_path)
```

Comment:
You can drop this print statement and I believe you can even drop the following line as it should be added to `PATH` automatically.

```suggestion

```

Reply:
Oh, really? Does that work with Conan v1 as well? I could have deleted so much dead code. :grin: 
Do you have an example of a recipe where `test_v1_package` works without one, maybe?